util-app/src/main/scala/com/twitter/app/App.scala

```diff
@@ -63,7 +63,7 @@ trait App extends Closable with CloseAwaitably {
 
   protected def exitOnError(reason: String): Unit = {
     System.err.println(reason)
-    close()
+    Await.result(close(), closeDeadline - Time.now)
```

Comment:
i think we should handle the failed `Future` case, otherwise the `System.exit` won't get called. also, i think something like:

`Await.result(close().unit, closeDeadline - Time.now)`